src/test/java/org/elasticsearch/search/indicesboost/SimpleIndicesBoostSearchTests.java

```diff
@@ -110,4 +100,207 @@ public void testIndicesBoost() throws Exception {
         assertThat(response.getHits().getAt(0).index(), equalTo("test2"));
         assertThat(response.getHits().getAt(1).index(), equalTo("test1"));
     }
+
+    @Test
+    public void testIndicesBoostWithAlias() throws Exception {
+        createIndex("alias_test1", "alias_test2");
+        ensureGreen();
+        client().admin().indices().prepareAliases().addAlias("alias_test1", "alias1").get();
+        client().admin().indices().prepareAliases().addAlias("alias_test2", "alias2").get();
+
+        client().index(indexRequest("alias_test1").type("type1").id("1")
+                .source(jsonBuilder().startObject().field("test", "value check").endObject())).actionGet();
+        client().index(indexRequest("alias_test2").type("type1").id("1")
+                .source(jsonBuilder().startObject().field("test", "value beck").endObject())).actionGet();
+        refresh();
+
+        float indexBoost = 1.1f;
+
+        logger.info("--- QUERY_THEN_FETCH");
+
+        logger.info("Query with alias_test1 boosted");
+        SearchResponse response = client().search(searchRequest()
+                        .searchType(SearchType.QUERY_THEN_FETCH)
+                        .source(searchSource().explain(true).indexBoost("alias1", indexBoost).query(termQuery("test", "value")))
+        ).actionGet();
+
+        assertThat(response.getHits().totalHits(), equalTo(2l));
+        logger.info("Hit[0] {} Explanation {}", response.getHits().getAt(0).index(), response.getHits().getAt(0).explanation());
+        logger.info("Hit[1] {} Explanation {}", response.getHits().getAt(1).index(), response.getHits().getAt(1).explanation());
+        assertThat(response.getHits().getAt(0).index(), equalTo("alias_test1"));
+        assertThat(response.getHits().getAt(1).index(), equalTo("alias_test2"));
+
+        logger.info("Query with alias_test2 boosted");
+        response = client().search(searchRequest()
+                        .searchType(SearchType.QUERY_THEN_FETCH)
+                        .source(searchSource().explain(true).indexBoost("alias2", indexBoost).query(termQuery("test", "value")))
+        ).actionGet();
+
+        assertThat(response.getHits().totalHits(), equalTo(2l));
+        logger.info("Hit[0] {} Explanation {}", response.getHits().getAt(0).index(), response.getHits().getAt(0).explanation());
+        logger.info("Hit[1] {} Explanation {}", response.getHits().getAt(1).index(), response.getHits().getAt(1).explanation());
+        assertThat(response.getHits().getAt(0).index(), equalTo("alias_test2"));
+        assertThat(response.getHits().getAt(1).index(), equalTo("alias_test1"));
+
+        logger.info("--- DFS_QUERY_THEN_FETCH");
+
+        logger.info("Query with alias_test1 boosted");
+        response = client().search(searchRequest()
+                        .searchType(SearchType.DFS_QUERY_THEN_FETCH)
```

Comment:
Right. I did it because existing test did that (I didn't dare to fix it).
I think I'll fix it.